16_React_Form/praktikum/src/components/MyForm.js

```diff
@@ -0,0 +1,134 @@
+import React, { useState, useRef } from "react";
+import "./MyForm.css";
+
+const MyForm = () => {
+  const emptyData = {
+    fullName: "",
+    email: "",
+    phoneNumber: "",
+    educationBg: "IT",
+    classCoding: "backend",
+    expectation: "",
+  };
+
+  const [data, setData] = useState(emptyData);
+  const [errorMsg, setErrorMsg] = useState();
+  const nameRegex = /^[a-zA-Z]+(([',. -][a-zA-Z ])?[a-zA-Z]*)*$/;
+  const emailRegex = /^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/;
+  const letter = useRef(null);
+
+  const handleInput = (e) => {
+    const name = e.target.name;
+    const value = e.target.value;
+
+    if (name === "fullName") {
+      if (nameRegex.test(value)) {
+        setErrorMsg("");
+      } else {
+        setErrorMsg("Full name must be character");
+      }
+    }
+
+    if (name === "email") {
+      if (emailRegex.test(value)) {
+        setErrorMsg("");
+      } else {
+        setErrorMsg("Email is not valid");
+      }
+    }
+
+    if (name === "phoneNumber") {
+      // Because index start from 0, so range {9-14} becomes {8-13}
+      if (data.phoneNumber.length <= 13 && data.phoneNumber.length >= 8) {
+        setErrorMsg("");
+      } else {
+        setErrorMsg("Phone Number is not valid");
+      }
+    }
```

Comment:
implementation of error message needs to implement as an object, each input item should have an error message handler in the state of error messages

example structure
```javascript
const initialErrorMsg = {
    fullName: "Full name should be a minimum of 3 characters",
    email: "Email is not valid",
    phoneNumber: "Phone number is not valid",
}
```
If the state of the error message is only a string, it will be difficult to handle the error for each input element.

Reply:
applied mas